### PR TITLE
flush queue when table change

### PR DESCRIPTION
**Pull request by roy-michael**

### CODE REVIEW

**Review by aroch — COMMENTED** [review]
Wouldn't it be a better approach to hold a writer per table?
If I understand correctly, In the worst-case scenario we have alternating events, each going to a different table, so we flush for batches of size 1.

File: src/main/java/ru/ivi/opensource/flinkclickhousesink/applied/ClickHouseSinkBuffer.java [hunk 1]
```diff
@@ -58,6 +58,13 @@ synchronized void tryAddToQueue() {
         }
     }
 
+    synchronized void tryAddToQueue(String table) {
```
[the remark this answers is not in the document]
Comment (roy-michael) [reply src/main/java/ru/ivi/opensource/flinkclickhousesink/applied/ClickHouseSinkBuffer.java:61]: synchronized was already there. i just added the table param

**Comment by roy-michael** [comment]
> Wouldn't it be a better approach to hold a writer per table?
> If I understand correctly, In the worst-case scenario we have alternating events, each going to a different table, so we flush for batches of size 1.

@aroch note that this is a short lived solution until we migrate the tenant schemas into a single one, and since the http client is async there is no need to have an actual writer per table.

**Comment by aroch** [comment]
It will take some time until we migrate to one table, but I get your point.

**Review by aroch — APPROVED** [review]

**Pull request merged**